### DISCUSSION TITLE
CIPP-API Integration - 401 Unauthorized

### DISCUSSION THREAD

**Discussion by perrytrienekens** [KelvinTegelaar/CIPP#1705] [Q&A]
Hi All,

We are trying to get the CIPP Public API working but ends up in an unexplained error. 

In Postman we end up with a 401 Unauthorized, while the token is configured
![image](https://github.com/KelvinTegelaar/CIPP/assets/47396034/1f21df24-583d-43c4-853d-b7bf429f899b)

In Powershell we tried the same and we retrieve the token successfully.
![image](https://github.com/KelvinTegelaar/CIPP/assets/47396034/ae93e861-2c56-4e3b-a92c-b18105223879)

More then happy to base on suggestions improve trouble shoot documentation on the API Page. As it's quite unclear where to debug this. The CIPP-API has been setup automatic. 

Only thing is we use Custom Domain for the CIPP app, maybe there is a hardcoded link somewhere? We changed already on API registration the URL but that was not solving the issue.

Has somebody activated the plugin and does it work? We are building an integration with Dynamics 365 for automatic user provisioning for all new user requests from end-customers, so great if CIPP-API will work. :)

Note: We also find out that everytime you "Disable/Enable Integration" it creates a new CIPP-API App registration, we ended with over 10+ :)

Thanks,

Perry

**Comment by KelvinTegelaar** [comment]
Not the place for support, if you want support you'll have to sponsor.
